docs/rules/no_sync_fn_in_async_fn.md

```diff
@@ -0,0 +1,37 @@
+Disallow sync function inside async function
+
+Using sync functions like `Deno.readTextFileSync` blocks the deno event loop so
+its not recommended to use it inside of an async function, because its stops
+progress of all other async tasks.
+
+### Invalid:
+
+```javascript
+async function foo() {
+  Deno.readTextFileSync("");
+}
+
+const fooFn = async function foo() {
+  Deno.readTextFileSync("");
+};
+
+const fooFn = async () => {
+  Deno.readTextFileSync("");
+};
+```
+
+### Valid:
+
+```javascript
+function foo() {
+  Deno.readTextFileSync("");
+}
+
+const fooFn = function foo() {
+  Deno.readTextFileSync("");
+};
+
+const fooFn = () => {
+  Deno.readTextFileSync("");
+};
+```
```

Comment:
It would be even better to also show valid examples with `async` functions containing async version of built-in functions, because we prefer async functions in general. Something like:

```ts
async function foo() {
  await Deno.readTextFile("");
}
```